py4cast/lightning.py

```diff
@@ -36,6 +42,62 @@
 PLOT_PERIOD: int = 10
 
 
+@dataclass
+class PlDataModule(pl.LightningDataModule):
+    """
+    DataModule to encapsulate data splits and data loading.
+    """
+
+    dataset: str
+    num_input_steps: int
+    num_pred_steps_train: int
+    num_pred_steps_val_test: int
+    dl_settings: TorchDataloaderSettings
+    dataset_conf: Union[Path, None] = None
+    config_override: Union[Dict, None] = (None,)
+
+    def __post_init__(self):
+        super().__init__()
+
```

Comment:
why do you call super's init here ? See the doc here: https://docs.python.org/3/library/dataclasses.html#dataclasses.__post_init__

Reply:
When I don't call super's init it returns an error, some methods such as "prepare_data_per_node" are not initilalized. 
I have read this in the doc :
The [__init__()] method generated by [@dataclass] does not call base class __init__() methods. If the base class has an __init__() method that has to be called, it is common to call this method in a [__post_init__()] method:
Note, however, that in general the dataclass-generated __init__() methods don’t need to be called, since the derived dataclass will take care of initializing all fields of any base class that is a dataclass itself. 
So I guess that base class is not a dataclass.

Reply:
thanks ! It seems "supported" by the dev team so okay :-)

https://github.com/Lightning-AI/pytorch-lightning/issues/8272